async_producer.go

```diff
@@ -610,6 +611,7 @@ func (p *asyncProducer) retryHandler() {
 
 func (p *asyncProducer) shutdown() {
 	Logger.Println("Producer shutting down.")
+	p.inFlight.Add(1)
```

Comment:
I don't think this is necessary? If you create a wait group and then immediately Wait on it (without ever calling Add) it just returns right away...